plugins/modules/packaging/os/slackpkg.py

```diff
@@ -67,12 +67,13 @@
 
 def query_package(module, slackpkg_path, name):
 
-    import glob
     import platform
+    import os
+    import re
 
     machine = platform.machine()
-    packages = glob.glob("/var/log/packages/%s-*-[%s|noarch]*" % (name,
-                                                                  machine))
+    pattern = re.compile('^%s-[^-]+-(%s|noarch)-[^-]+$' % (name, machine))
+    packages = [f for f in os.listdir('/var/log/packages') if re.match(pattern, f)]
```

Comment:
Since you already compiled it:
```suggestion
    packages = [f for f in os.listdir('/var/log/packages') if pattern.match(f)]
```